pkg/controller/direct/sql/merge.go

```diff
@@ -0,0 +1,906 @@
+			if existingMysqlReplicaConfig {
+				if direct.ValueOf(desired.Spec.ReplicaConfiguration.Password.Value) != actual.ReplicaConfiguration.MysqlReplicaConfiguration.Password {
+					// Change password
+					updateRequired = true
+				}
+			} else {
+				// Add password
+				updateRequired = true
+			}
+			merged.ReplicaConfiguration.MysqlReplicaConfiguration.Password = direct.ValueOf(desired.Spec.ReplicaConfiguration.Password.Value)
+		} else if existingMysqlReplicaConfig && actual.ReplicaConfiguration.MysqlReplicaConfiguration.Password != "" {
+			// Remove password
+			updateRequired = true
+		}
+
+		if desired.Spec.ReplicaConfiguration.SslCipher != nil {
+			if existingMysqlReplicaConfig {
+				if direct.ValueOf(desired.Spec.ReplicaConfiguration.SslCipher) != actual.ReplicaConfiguration.MysqlReplicaConfiguration.SslCipher {
+					// Change SSL cipher
+					updateRequired = true
+				}
+			} else {
+				// Add SSL cipher
+				updateRequired = true
+			}
+			merged.ReplicaConfiguration.MysqlReplicaConfiguration.SslCipher = direct.ValueOf(desired.Spec.ReplicaConfiguration.SslCipher)
+		} else if existingMysqlReplicaConfig && actual.ReplicaConfiguration.MysqlReplicaConfiguration.SslCipher != "" {
+			// Remove SSL cipher
+			updateRequired = true
+		}
+
+		if desired.Spec.ReplicaConfiguration.Username != nil {
+			if existingMysqlReplicaConfig {
+				if direct.ValueOf(desired.Spec.ReplicaConfiguration.Username) != actual.ReplicaConfiguration.MysqlReplicaConfiguration.Username {
+					// Change username
+					updateRequired = true
+				}
+			} else {
+				// Add username
+				updateRequired = true
+			}
+			merged.ReplicaConfiguration.MysqlReplicaConfiguration.Username = direct.ValueOf(desired.Spec.ReplicaConfiguration.Username)
+		} else if existingMysqlReplicaConfig && actual.ReplicaConfiguration.MysqlReplicaConfiguration.Username != "" {
+			// Remove username
+			updateRequired = true
+		}
+
+		if desired.Spec.ReplicaConfiguration.VerifyServerCertificate != nil {
+			if existingMysqlReplicaConfig {
+				if direct.ValueOf(desired.Spec.ReplicaConfiguration.VerifyServerCertificate) != actual.ReplicaConfiguration.MysqlReplicaConfiguration.VerifyServerCertificate {
+					// Change verify server certificate
+					updateRequired = true
+				}
+			} else {
+				// Add verify server certificate
+				updateRequired = true
+			}
+			merged.ReplicaConfiguration.MysqlReplicaConfiguration.VerifyServerCertificate = direct.ValueOf(desired.Spec.ReplicaConfiguration.VerifyServerCertificate)
+		} else if existingMysqlReplicaConfig && actual.ReplicaConfiguration.MysqlReplicaConfiguration.VerifyServerCertificate {
+			// Remove verify server certificate
+			updateRequired = true
+		}
+	} else if existingReplicaConfig {
+		// Remove replica configuration
+		updateRequired = true
+	}
+
+	if desired.Spec.RootPassword != nil && desired.Spec.RootPassword.Value != nil {
+		if direct.ValueOf(desired.Spec.RootPassword.Value) != actual.RootPassword {
```

Comment:
Hmm, do you know if we have a pattern already for how to handle "mutable but unreadable" fields in SciFi? It will probably be difficult to tell when the "mutable but unreadable" fields are changed, and therefore we wouldn't necessarily know when to make a GCP API call to update the resource. If possible, I would like to avoid having to make an API call every re-reconcile "just in case" the field is changed.

Reply:
Discussed offline, this will require a separate PR to handle "mutable but unreadable" fields for SQLInstance